File: Python/programmers/etc/level1/최대공약수와_최소공배수.py
# programmers는 gcd, lcm을 쓸 수가 없다.
def gcd(n, m):
    while m:
        n, m = m, n%m
    return n

def solution(n, m):
    return [gcd(n,m), n*m//gcd(n,m)]


n, m = 3, 12
print(solution(n, m)) # [3, 12]

n, m = 2, 5
print(solution(n, m)) # [1, 10]

File: Java/programmers/etc/level1/최대공약수와최소공배수.java
package CodeTest.Java.programmers.etc.level1;

public class 최대공약수와최소공배수 {
    public int[] solution(int n, int m) {
        return new int[]{gcd(n,m), n*m/gcd(n,m)};
    }
    
    private int gcd(int n, int m) {
        while (m != 0) {
            int tmp = n;
            n = m;
            m = tmp%m;
        }
        return n;
    }
}
